src/components/Autocomplete.tsx

```diff
@@ -0,0 +1,117 @@
+import { useCallback, useRef, useState } from 'react';
+import debounce from 'lodash.debounce';
+
+import './Autocomplete.scss';
+import { Person } from '../types/Person';
+
+type Props = {
+  people: Person[];
+  onSelected?: (person: Person | null) => void;
+  delay: number;
+};
+
+export const Autocomplete: React.FC<Props> = ({
+  people,
+  onSelected = () => {},
+  delay,
+}) => {
+  const queryField = useRef<HTMLInputElement | null>(null);
+
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [focusInput, setFocusInput] = useState(false);
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, delay), []);
```

Comment:
is it required to memoize this function?

Reply:
Yes. If I don't, the components will be rendering every time when I press a key and the suggestions will appear after some delay, but also rendering again several times, not once.